hector_gazebo_plugins/include/hector_gazebo_plugins/gazebo_ros_magnetic.h

```diff
@@ -85,6 +91,9 @@ class GazeboRosMagnetic : public ModelPlugin
   event::ConnectionPtr updateConnection;
 
   boost::shared_ptr<dynamic_reconfigure::Server<SensorModelConfig> > dynamic_reconfigure_server_;
+
+  void Update_Vector3Stamped(common::Time &sim_time);
+  void Update_MagneticField(common::Time &sim_time);
```

Comment:
I'm not sure why you introduce separate methods for that but if you do `sim_time` should be a const reference as it's not modified.
I also don't think you should follow the lead of the existing code and pass parameters such as the message or the field vector as member variables.

Reply:
I split it out into separate functions to cut down on code-duplication & to hopefully make the code a little simpler.  Otherwise it would have been a lot of nested ifs & macros, which just felt messy to read.

I've made the sim_time const in the latest revision.

Reply:
Those functions are used once, so, there is no reduction of code-duplication, and having the code in place would allow `field_` to be used without passing it as a member variable which is bad style (same holds for `magnetic_field_old_` and `magnetic_field_new_`).
I know in this case it's pretty safe but passing variables as members can go horribly wrong if multithreading gets involved so it's better to avoid it generally.

Reply:
Refactored to remove the Update_Vector3Stamped and Update_MagneticField functions; everything is back in Update now.  The magnetic_field_new_ and magnetic_field_old_ members are also removed, and the magnetic field message (whether a Vector3Stamped or MagneticField) is declared as a local variable within Update.

Reply:
Thank you for the update and your work on this!